src/acm_agents.py

```diff
@@ -0,0 +1,268 @@
+
+from IPython.display import Image, display
+from langchain_core.runnables.graph import CurveStyle, MermaidDrawMethod
+from langchain_openai import ChatOpenAI
+
+
+
+
+
+
+class AgentState(TypedDict):
+    #messages: Annotated[list[AnyMessage], operator.add]
+    messages: Annotated[Sequence[BaseMessage], add_messages]
+    task: str
+    route:str
+    content: str
+    iteration: int
+    score:int
+    summary: str
+    pr: str
+    srch:str
+
+ROUTER_PROMPT = """You are an expert engineer of Red Hat Advanced Cluster Management (RHACM or ACM). 
+You will see the question and decide how to route it. 
+If the question is about creating a ACM GRC Policy, then it will routed to the author_node.
+Any other questions should be routed to the search_node. 
+Therefore you will return one of the choices mentioned below as per your understanding. 
+- author_node 
+- search_node. 
+Do not attempt to return any other things. """
+
+AUTHOR_PROMPT = """You are an expert at writing GRC (governance risk and compliance) Policies 
+for Red Hat Advanced Cluster Management (RHACM or ACM). 
+You are aware of the different kind (CRDs) under apiVersion: policy.open-cluster-management.io 
+and know how to use them.
+You can write a ACM policy yaml given a task as below 
+If a user gives you some feedback on the yaml you have produced,
+process it, think through it and improve the yaml. 
+And clarify each of the changes that you have made to improve.
+------
+
+{content}"""
+
+CRITIC_PROMPT = """You are an expert at testing GRC (governance risk and compliance) Policies 
+for Red Hat Advanced Cluster Management (RHACM or ACM). 
+You are aware of the different kind (CRDs) under apiVersion: policy.open-cluster-management.io 
+and know how to use them.
+Given a policy yaml as below, you can find out the flaws in it and suggest the changes to be made point by point.
+------
+
+{content}"""
+
+SCORER_PROMPT = """You are an expert evaluator of GRC (governance risk and compliance) Policies 
+for Red Hat Advanced Cluster Management (RHACM or ACM). 
+You are aware of the different kind (CRDs) under apiVersion: policy.open-cluster-management.io 
+and know how to use them.
+Rate the given content on a scale of 0-100 based on: 
+    - Technical accuracy (50 points) 
+    - Completeness (50 points) 
+Provide only the numerical score as your response.
+-------
+
+{content}"""
+
+PR_PROMPT = """In the content provided to you, there is a  GRC (governance risk and compliance) Policy 
+for Red Hat Advanced Cluster Management (RHACM or ACM) definition buried.
+Just output that in a nicely formatted fashion
+------
+
+{content}"""
+
+root_nodes = ['author','search']
+
+
+_ = load_dotenv(find_dotenv()) 
+openai.api_key  = os.getenv('OPENAI_API_KEY')
+#model = ChatOpenAI(model="gpt-3.5-turbo", temperature=0)
+#gpt-4o
+#gpt-4o-mini
+#gpt-3.5-turbo
+#o1
+#o3-mini
+model = ChatOpenAI(model="gpt-4o", temperature=0)
```

Comment:
Can the demo run with the groq client/model? 

I'd love to try it with your workflow - it's really attractive!